internal/connect/system.go

```diff
@@ -3,18 +3,34 @@ package connect
 import (
 	"bytes"
 	"fmt"
+	"io"
 	"os"
 	"os/exec"
 	"strings"
 )
 
+var systemEcho bool
+
+// SetSystemEcho toggles piping of executed command's outputs to stdout/stderr
+// returns true if it was enabled before, false otherwise
+func SetSystemEcho(v bool) bool {
+	prev := systemEcho
+	systemEcho = v
+	return prev
+}
+
 // Assign function for running external commands to a variable so it can be mocked by tests.
 var execute = func(cmd []string, validExitCodes []int) ([]byte, error) {
 	Debug.Printf("Executing: %s\n", cmd)
 	var stderr, stdout bytes.Buffer
 	comm := exec.Command(cmd[0], cmd[1:]...)
-	comm.Stdout = &stdout
-	comm.Stderr = &stderr
+	if systemEcho {
```

Comment:
so maybe we could add this here
`fmt.Fprintf(os.Stdout,"\nExecuting '%s'\n\n", strings.Join(cmd, " "))`

Then there would be no need for the `QuietOut.Printf` in zypper.go, and the QuietOut logger could be removed from the connect package, as it's specific to the zypper plugin

Reply:
hmm... you mean to print it only when `systemEcho` is enabled?

Reply:
yes

Reply:
ok